src/coreclr/nativeaot/Runtime/arm64/WriteBarriers.asm

```diff
@@ -320,7 +315,6 @@ NotInHeap
         b      DoCardsCmpXchg
 CmpXchgRetry
         ;; Check location value is what we expect.
-    ALTERNATE_ENTRY  RhpCheckedLockCmpXchgAVLocation2
         ldaxr   x10, [x0]
```

Comment:
The `RhpCheckedLockCmpXchgAVLocation2` and `RhpCheckedXchgAVLocation2` checks are weird. They seem to be present for the same location that was already null checked and unnecessary.